cmd/utils/flags.go

```diff
@@ -1457,11 +1457,20 @@ func setEthashDatasetDir(ctx *cli.Context, cfg *eth.Config) {
 	}
 }
 
-func setEthash(ctx *cli.Context, cfg *eth.Config) {
-	if ctx.GlobalIsSet(EthashCacheDirFlag.Name) {
+func setEthashCacheDir(ctx *cli.Context, cfg *eth.Config) {
+	switch {
+	case ctx.GlobalIsSet(EthashCacheDirFlag.Name):
 		cfg.Ethash.CacheDir = ctx.GlobalString(EthashCacheDirFlag.Name)
+
+	case (ctx.GlobalBool(ClassicFlag.Name) || ctx.GlobalBool(MordorFlag.Name)) && cfg.Ethash.CacheDir == eth.DefaultConfig.Ethash.CacheDir:
```

Comment:
Is `&& cfg.Ethash.CacheDir == eth.DefaultConfig.Ethash.CacheDir` necessary? Is the only way to set this value by using `EthashCacheDirFlag`?

Reply:
Ah, in `setEthash` utils function?

Reply:
I'm not sure if there are any cases where it would be non defaulty, this pattern was copied from the existing setDataDir function.

Reply:
Roger. I don't see any harm in an extra check, anyway.